devserver/devserver/devserver.go

```diff
@@ -175,34 +169,40 @@ func CreateFileHandler(servingPath, manifest string, pkgs []string, base string)
 
 // dirHTTPFileSystem implements http.FileSystem by looking in the list of dirs one after each other.
 type dirHTTPFileSystem struct {
-	files []string
+	packageDirs []string
 	base string
 }
 
 func (fs dirHTTPFileSystem) Open(name string) (http.File, error) {
-	for _, packageName := range fs.files {
-		filePackageName := filepath.Join(packageName, name)
-		realFilePath, err := runfiles.Runfile(fs.base, filePackageName)
-		stat, statErr := os.Stat(realFilePath)
+	for _, packageName := range fs.packageDirs {
+		manifestFilePath := filepath.Join(packageName, name)
+		realFilePath, err := runfiles.Runfile(fs.base, manifestFilePath)
 
-		if err != nil || statErr != nil {
+		if err != nil {
 			// In case the runfile could not be found, we also need to check that the requested
 			// path does not refer to a directory containing an "index.html" file. This can
 			// happen if Bazel runs without runfile symlinks, where only files can be resolved
 			// from the manifest. In that case we dirty check if there is a "index.html" file.
-			realFilePath, err = runfiles.Runfile(fs.base, filepath.Join(filePackageName, "index.html"))
+			realFilePath, err = runfiles.Runfile(fs.base, filepath.Join(manifestFilePath, "index.html"))
 
-			// Continue searching if the runfile couldn't be found for the request filed.
-			if _, statErr := os.Stat(realFilePath); err != nil || statErr != nil {
+			// Continue searching if the runfile couldn't be found for the requested file.
+			if err != nil {
 				continue
 			}
 		}
 
+		stat, err := os.Stat(realFilePath)
```

Comment:
Internally `Runfile` checks for the existence of the file. I think this check is redundant.